### PR TITLE
#16228. Adding endpoint for retrieving accounts affiliated filtered b…

### PR DESCRIPTION
**Pull request by hfekete**
*Issue #16228:*
https://github.com/bcgov/entity/issues/16228

*Description of changes:*
- Adding endpoint for retrieving accounts affiliated filtered by business identifier
- Added UUID to Org to be able to publicly and uniquely identify account.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/models/org.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ class Org(VersionedModel):  # pylint: disable=too-few-public-methods,too-many-in
     __tablename__ = 'orgs'
 
     id = Column(Integer, primary_key=True)
+    uuid = Column(UUID, nullable=False, server_default=text("uuid_generate_v4()"), unique=True)
```
Comment (seeker25) [inline auth-api/src/auth_api/models/org.py:47]: What's the purpose of creating a uuid, if we already have id?

File: auth-api/src/auth_api/resources/org.py [hunk 2]
```diff
@@ -393,6 +393,40 @@ def get(org_id):
...
+@API.route('/affiliation/<string:business_identifier>', methods=['GET', 'OPTIONS'])
+class OrgInfoByAffiliation(Resource):
+    @staticmethod
+    @TRACER.trace()
+    @cors.crossdomain(origin='*')
+    @_jwt.has_one_of_roles(
+        [Role.SYSTEM.value, Role.STAFF_VIEW_ACCOUNTS.value, Role.PUBLIC_USER.value])
+    def get(business_identifier):
+        """Search orgs by BusinessIdentifier and return org Name and UUID."""
+        org_search = OrgSearch(
```
Comment (seeker25) [inline auth-api/src/auth_api/resources/org.py:406]: I'd recommend changing the OrgSearch dataclass to have optional fields?

File: auth-api/src/auth_api/resources/org.py [hunk 3]
```diff
@@ -393,6 +393,40 @@ def get(org_id):
...
+            "",
+            "",
+            "",
+            "",
+            int(request.args.get('page', 1)),
+            int(request.args.get('limit', 10))
+        )
+
+        try:
+            data = OrgService.search_orgs(org_search)
```
Comment (seeker25) [inline auth-api/src/auth_api/resources/org.py:421]: Why are we looking for orgs on the affiliation endpoint?

File: auth-api/tests/unit/api/test_org.py [hunk 4]
```diff
@@ -2246,3 +2246,58 @@ def test_get_org_affiliations(client, jwt, session, keycloak_mock, mocker,
...
+        dictionary = json.loads(rv.data)
+        org_id = dictionary['id']
+
+        client.post('/api/v1/orgs/{}/affiliations'.format(org_id), headers=headers,
+                    data=json.dumps(TestAffliationInfo.affiliation3), content_type='application/json')
+
+    return created_orgs
+
+
+@pytest.mark.parametrize("expected_http_status, entries_count",
```
Comment (seeker25) [inline auth-api/tests/unit/api/test_org.py:2273]: Nice job here, I appreciate the unit tests!

**Comment by kialj876** [comment]
Great work! Just need to fix the linter (check out the makefile to see how its run in the PR). Other than that nothing to add on to Travis's review, looks good 👍

File: auth-api/src/auth_api/resources/org.py (hunk 2 above)
Comment (hfekete) [reply auth-api/src/auth_api/resources/org.py]: I added new service call, new call in model and updated the endpoint call. Let me know if it is better now. 

Thanks!

File: auth-api/src/auth_api/models/org.py [hunk 5]
```diff
@@ -150,6 +157,19 @@ def search_org(cls, search: OrgSearch):
 
         return pagination.items, pagination.total
 
+    @classmethod
+    def search_orgs_by_business_identifier(cls, business_identifier, pagination_info: PaginationInfo):
```
Comment (seeker25) [inline auth-api/src/auth_api/models/org.py:161]: Excellent

**Review by seeker25 — APPROVED** [review]

**Review by kialj876 — APPROVED** [review]

**Comment by seeker25** [comment]
Side note: 

Migrations user didn't have permissions to add the extension CREATE EXTENSION IF NOT EXISTS "uuid-ossp";

I've manually went onto the DEV/TEST/PROD environments and done this.

**Pull request merged**